src/listeners/e2e.listener.ts

```diff
@@ -0,0 +1,18 @@
+import { DAPP_ACTIONS } from '../constants/dapp-actions.enum'
+import { Message } from '../messaging/scripts.messaging'
+import { isBackgroundAction } from '../utils/asserts'
+import { messageHandler } from './message-listeners'
+
+chrome.runtime.onMessageExternal.addListener((request: Message<unknown>, sender, sendResponse) => {
+  const { action } = request || {}
+
+  if (!isBackgroundAction(action) || !DAPP_ACTIONS.includes(action)) {
+    sendResponse({ error: `Blossom: invalid action received from the extension with ID ${sender.id}` })
+
+    return true
+  }
+
+  messageHandler(request, sender, sendResponse)
```

Comment:
maybe it could have a separate message handler from the content event listener to clearly check the permissions to do actions.

Reply:
This handler also checks this enum https://github.com/fairDataSociety/blossom/blob/login/src/constants/dapp-actions.enum.ts for allowed actions. So it allows the same actions as the ones exposed to dapps. But maybe we could separate them into two enums, one for extensions and one for dapps?

Reply:
yes please, because dApps shouldn't call the same actions like a extension's popup page. see above my comment regarding to this.

Reply:
Sure, definitely we want to limit actions that can be invoked by dapps. But what I meant was that this enum is used to check what actions are allowed to be invoked by dapps and other extensions that use e2e API. For other Blossom components, all actions are available.